src/FINDOLOGIC/FindologicApi.php

```diff
@@ -0,0 +1,276 @@
+<?php
+
+namespace FINDOLOGIC;
+
+use FINDOLOGIC\Definitions\RequestType;
+use FINDOLOGIC\Exceptions\ConfigException;
+use FINDOLOGIC\Exceptions\ParamNotSetException;
+use FINDOLOGIC\Exceptions\ServiceNotAliveException;
+use FINDOLOGIC\Helpers\ParameterBuilder;
+use FINDOLOGIC\Objects\JsonResponse;
+use FINDOLOGIC\Objects\XmlResponse;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+
+class FindologicApi extends ParameterBuilder
```

Comment:
As commented below as well, consider turning this into a request factory, ie. provide something like a `createSearchRequest()` method and have different classes for the different request types. The different request classes would be aware of their required parameters, and a common ancestor class would define shared required parameters, eg. `shopkey`; request instances could have their own timing info. All in all this seems like a good semantic match for me.

Reply:
Created #16.